test/freenet/support/PrioritizedTickerTest.java

```diff
@@ -55,6 +67,71 @@ public void testSimple() throws InterruptedException {
 		}
 	}
 
+    public void testRemove() throws InterruptedException {
```

Comment:
Don’t take this the wrong way but this test is horrible.

* It is unclear what is being tested.
* Multiple things seem to be tested at once. It’s better to have multiple tests that only test a single thing each.
* `Thread.sleep` is used. This will unnecessarily delay the test and make test execution slower for everyone else. It’s better to mock the part that requires the passage of time so that all required actions can run as fast as possible.
* You rely on an assert. It’s better to use special exceptions if you rely on the behaviour. If you only want to make sure that certain things happened, a verifying mock would be more appropriate.
* The test relies on certain conditions that it painstakingly tries to create itself. Again, a mock that is simply configured to match that condition would make this a lot clearer.